src/test/ui/did_you_mean/issue-40006.rs

```diff
@@ -1,16 +1,22 @@
-impl dyn X { //~ ERROR cannot be made into an object
-//~^ ERROR missing
+impl dyn A { //~ ERROR missing
     Y
 }
 
 struct S;
 
-trait X { //~ ERROR missing
+trait A { //~ ERROR missing
```

Comment:
I would prefer not altering this file to see how the new logic affects the file as it was and because this is a regression test. Can you make a new file?

Reply:
Originally was going to do so, but this test is only checking that the error recovery actually recovers gracefully from all of these cases. The prior logic tried its best to recover inside of the block so the method `xxx` for example would actually end up in the trait's AST. The new logic just swallows everything until the block is completed. This causes knock down errors for unused items of method not found, but I feel that is better than the current state where a single missing close brace early in the file can cause lots of parse errors unnecessarily. I'll see about making some changes to avoid the spurious errors from this, but regardless this file without changes would just hide everything after the second error, not really testing much :-/

Reply:
Well suppose that we change the strategy two years after this PR... Isn't it a good idea then to have more variations to see what effect a new PR has on different inputs?

Reply:
Done. Added both in the same file.